src/test/java/org/opensearch/security/dlic/dlsfls/DfmOverwritesAllTest.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright OpenSearch Contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package org.opensearch.security.dlic.dlsfls;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.support.WriteRequest.RefreshPolicy;
+import org.opensearch.client.Client;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.test.DynamicSecurityConfig;
+import org.opensearch.security.test.helper.rest.RestHelper.HttpResponse;
+
+/**
+ * Tests that the dfm_empty_overwrites_all flag works correctly.
+ * Per default, if a user has a role that adds restrictions to an index
+ * regarding DLS, FLS or masked fields (DFM), and another role that has no restrictions
+ * on the same index, the restrictions from the first role still applies.
+ *
+ * If the dfm_empty_overwrites_all flag is set to true, the logic is reversed:
+ * If a user has a role that places no restrictions on an index, this trumps
+ * all other role that eventually do place restrictions on this index.
+ */
+public class DfmOverwritesAllTest extends AbstractDlsFlsTest {
```

Comment:
Can we add unit tests in addition to these integration tests - I am unable to interpret if the assertions in these tests are correct behavior or not.

Reply:
I do not see any comments or updates in this issue

Reply:
There're unit tests. The comments of each test method explain the test cases.

Reply:
These tests integration tests, because they test multiple systems working together - spinning up a OpenSearch cluster.  Unit tests would as example invoke the method `getDlsFls(...)` valid its outcome directly.  The value of unit tests is they are much faster to execute and can validate error cases easier.

I will not block this PR on a practice that is not widely adopted in this codebase - as I look around I mostly see integration tests.  Keep the validation strategy in mind as we continue adding improvements so we can ensure a maintainable codebase.